src/Stratis.Bitcoin.Features.Consensus/Rules/CommonRules/StraxCoinstakeRule.cs

```diff
@@ -80,6 +80,13 @@ public override Task RunAsync(RuleContext context)
                     // Only OP_RETURN scripts are allowed in coinbase.
                     if (!txOut.ScriptPubKey.IsUnspendable)
                     {
+                        // Make an exception for refund outputs.
+                        if (context.ValidationContext.ChainedHeaderToValidate.Header is PosBlockHeader posHeader && posHeader.HasSmartContractFields)
```

Comment:
maybe unite 2 ifs into one

Reply:
@rowandh , I think we need a better check here to deal with refund outputs.